service/matching/tasklist/task_list_manager.go

```diff
@@ -263,6 +317,7 @@ func (c *taskListManagerImpl) AddTask(ctx context.Context, params AddTaskParams)
 		c.Stop()
 		return false, errShutdown
 	}
+	c.reloadTaskListPartitionConfig()
```

Comment:
why do we need to call this here if it's already called at Start?

Reply:
it could fail at Start, because non-root partition may be created before the root partition given current setting.